tests/dspy/test_dspy_autolog.py

```diff
@@ -579,3 +578,139 @@ def test_autolog_log_compile(log_compiles):
         assert "best_model.json" in artifacts
     else:
         assert mlflow.last_active_run() is None
+
+
+def test_autolog_log_nested_compile():
+    class NestedOptimizer(dspy.teleprompt.Teleprompter):
+        def compile(self, program):
+            callback = dspy.settings.callbacks[0]
+            assert callback.optimizer_stack_level == 2
+            return program
+
+    class DummyOptimizer(dspy.teleprompt.Teleprompter):
+        def __init__(self):
+            super().__init__()
+            self.nested_optimizer = NestedOptimizer()
+
+        def compile(self, program):
+            self.nested_optimizer.compile(program)
+            callback = dspy.settings.callbacks[0]
+            assert callback.optimizer_stack_level == 1
+            return program
+
+    mlflow.dspy.autolog(log_compiles=True)
+    dspy.settings.configure(lm=DummyLM([{"answer": "4", "reasoning": "reason"}]))
+
+    program = dspy.ChainOfThought("question -> answer")
+    optimizer = DummyOptimizer()
+
+    optimizer.compile(program)
+
+    assert dspy.settings.callbacks[0].optimizer_stack_level == 0
+    run = mlflow.last_active_run()
+    assert run is not None
+    client = MlflowClient()
+    artifacts = (x.path for x in client.list_artifacts(run.info.run_id))
+    assert "best_model.json" in artifacts
+
+
+@pytest.mark.skipif(
+    Version(importlib.metadata.version("dspy")) < Version("2.6.12"),
+    reason="evaluate callback is available since 2.6.12",
+)
```

Comment:
```suggestion
skip_if_callback_unavailable = pytest.mark.skipif(
    Version(importlib.metadata.version("dspy")) < Version("2.6.12"),
    reason="evaluate callback is available since 2.6.12",
)
@skip_if_callback_unavailable
```